src/scenarios/personalization/ProgressSet/ProgressSet.tsx

```diff
@@ -0,0 +1,269 @@
+import React, { useState, useRef, useEffect, useContext } from 'react';
+import {
+  Button,
+  Container,
+  Divider,
+  Form,
+  Grid,
+  Header,
+  Label,
+  Segment,
+} from 'semantic-ui-react';
+import {
+  useScenarioHost,
+  VariableSearch,
+} from '@axinom/mosaic-fe-samples-host';
+import { getApolloClient } from '../../../apollo-client';
+import { SetProgressMutation } from './graphql-documents';
+
+const SharedState = React.createContext(false);
+
+export default SharedState;
+
+export const ProgressSet: React.FC = () => {
+  const [userAccessToken, setUserAccessToken] = useState<string>();
+  const [frequencyOfProgressSaving, setFrequencyOfProgressSaving] =
+    useState<number>();
+  const [key, setKey] = useState<string>();
+  const [isRunning, setIsRunning] = useState<boolean>(false);
+
+  const setProgress = async (isRunning: boolean): Promise<void> => {
```

Comment:
Looks like the purpose of this function is to toggle the value of `isRunning`? If so, i think it makes more sense to name it as `toggleIsRunning`. Also we don't need `isRunning` parameter, and the function does not need to be async (no need to return a promise either).

eg: 

```
const toggleIsRunning = (): void => {
    setIsRunning(prevIsRunning => !prevIsRunning);
  };
```

Since it's just a one liner and also the fact that this function is called from only one place, consider removing this function altogether and using `setIsRunning(prevIsRunning => !prevIsRunning);` in line 95